src/validation/custom.validation.ts

```diff
@@ -91,22 +91,61 @@ export const checkAtLeastOneFieldHasValue = (errMsg: string, ...fields: any[]) =
   throw new Error(errMsg);
 };
 
-export const checkMandatoryTrustFields = (nameErrMsg, dateErrMsg, trustsJson: string) => {
+export const checkMandatoryTrustFields = (trustsJson: string) => {
   const trusts: trustType.Trust[] = JSON.parse(trustsJson);
+  const addressMaxLength = 50;
+
   for (const trust of trusts) {
-    if (
-      trust.creation_date_day === undefined ||
-      trust.creation_date_day === "" ||
-      trust.creation_date_month === undefined ||
-      trust.creation_date_month === "" ||
-      trust.creation_date_year === undefined ||
-      trust.creation_date_year === ""
-    ) {
-      throw new Error(dateErrMsg);
+    checkTrustCreationDate(trust);
+
+    checkTrustName(trust);
+
+    checkIndividualsAddress(trust, addressMaxLength);
+
+    checkCorporatesAddress(trust, addressMaxLength);
+  }
+  return true;
+};
+
+const checkTrustCreationDate = (trust: trustType.Trust) => {
+  if (trust.creation_date_day === undefined ||
+    trust.creation_date_day === "" ||
+    trust.creation_date_month === undefined ||
+    trust.creation_date_month === "" ||
+    trust.creation_date_year === undefined ||
+    trust.creation_date_year === "") {
+    throw new Error(ErrorMessages.TRUST_CREATION_DATE);
+  }
+};
+
+const checkTrustName = (trust: trustType.Trust) => {
+  if (trust.trust_name === undefined || trust.trust_name === "") {
+    throw new Error(ErrorMessages.TRUST_NAME);
+  }
+};
+
+const checkIndividualsAddress = (trust: trustType.Trust, addressMaxLength: number) => {
+  if (trust.INDIVIDUALS) {
+    for (const individual of trust.INDIVIDUALS) {
+      if (individual.ura_address_premises?.length > addressMaxLength) {
```

Comment:
If this is a mandatory field (as name of parent method suggests) is the '?' necessary here? It's not present in the name and date 'check' methods.

Reply:
Checked with Mike Moore, those fields can be missing (ie optional). Will mark them as optional in the model and add extra test to allow missing values for those fields.